client/src/components/Patient/PatientManageMedicalRecords.tsx

```diff
@@ -0,0 +1,427 @@
+import React, { FunctionComponent, useContext, useState ,useEffect} from 'react';
+import ReactPaginate from 'react-paginate';
+import { HealthContext } from '../../providers/HealthProvider';
+import { AuthContext } from '../../providers/AuthProvider';
+import AWS from 'aws-sdk';
+
+const PatientManageMedicalRecords: FunctionComponent<{}> = () => {
+  const { user, role, logout } = useContext(AuthContext);
+  const { getAllRequestsForPatient,approveMedicalRecordsRequestHealthBlockContract,rejectMedicalRecordsRequestHealthBlockContract, currentAccount} = useContext(HealthContext);
+  const [currentPage, setCurrentPage] = useState(1);
+  const[medicalRecordRequests,setMedicalRecordRequests]=useState<any>([]);
+  const [filterOption, setFilterOption] = useState<string>('all');
+  const [showDeclinePopup, setShowDeclinePopup] = useState(false);
+  const [selectedFile, setSelectedFile] = useState<undefined | File>(undefined);
+  const [showAcceptPopup, setShowAcceptPopup] = useState(false);
+
+  const GetAllMedicalRecordRequests = async (patientAddress: string) => {
+    try {
+      const requests=await getAllRequestsForPatient?.(patientAddress);
+      console.log(requests);
+      setMedicalRecordRequests(requests);
+    } catch (err) {
+      console.log("Error while fetching the medical Record Requests",err);
+    }
+  };
+
+  const approveRequest=async(requestID: string)=>{
+    try{
+      await approveMedicalRecordsRequestHealthBlockContract?.(requestID)
+      if(currentAccount){
+        GetAllMedicalRecordRequests(currentAccount);}
+    }catch(err){
+      console.log("Error while approving the medical Record Requests",err);
+    }
+  };
+  const rejectRequest=async(requestID: string)=>{
+    try{
+      await rejectMedicalRecordsRequestHealthBlockContract?.(requestID)
+      if(currentAccount){
+      GetAllMedicalRecordRequests(currentAccount);}
```

Comment:
why does this exists? where is the result stored?

Reply:
To get all the medical record requests list. The response is being set to medicalRecordsRequests state to display the medical records.